Sources/TuistSupport/System/System.swift

```diff
@@ -260,19 +210,26 @@ public final class System: Systeming {
         try localFileSystem.chmod(mode, path: path, options: options)
     }
 
-    private func runAndPrint(
+    public func runAndPrint(
         _ arguments: [String],
-        verbose: Bool,
-        environment: [String: String],
-        redirection: TSCBasic.Process.OutputRedirection
+        verbose: Bool = false,
+        environment: [String: String] = ProcessInfo.processInfo.environment,
+        redirection: TSCBasic.Process.OutputRedirection = .none
     ) throws {
+        // We have to collect both stderr and stdout because we want to stream the output
+        // the `ProcessResult` type will not contain either unless outputRedirection is set to `.collect`
+        var errorData: [UInt8] = []
+        var stdOutData: [UInt8] = []
+
         let process = Process(
             arguments: arguments,
             environment: environment,
             outputRedirection: .stream(stdout: { bytes in
+                stdOutData.append(contentsOf: bytes)
```

Comment:
One potential issue with doing this by default, is that we might overflow the memory with processes like `xcodebuild`'s, which outputs a lot of data. What if we make this controllable by the caller?

Reply:
For now i can just remove it, i added it for completeness.  I'll take it into account with the refactor i mentioned in Slack.